### PR TITLE
update from foglamp bad bearing code

### PR DESCRIPTION
**Pull request by doug-dianomic**
. clean up spacing
. in serve, handle EOF and exception on initial read separately for more graceful exit/management
. handle SystemExit for when server is bailing
. method getattr, make argument optional

### CODE REVIEW

File: python/fledge/common/iprpc.py [hunk 1]
```diff
@@ -288,18 +308,20 @@ def log_errors(fd):
...
             # Wait until process terminates (without using p.wait())
             if p.poll() is not None:
-                _LOGGER.error("{} module load exited with return code {}".format(name, p.returncode))
+                _LOGGER.error("{} module load exited with return code {}".format(server_args, p.returncode))
                 raise RuntimeError(p.returncode)
             # Process hasn't exited yet, let's wait some
             time.sleep(0.5)
 
-        # special protocol, now tell the server the name of the mapped arg file
+        # special prtocol, now tell the server the name of the mapped arg file
```
Comment (praveen-garg) [inline python/fledge/common/iprpc.py:321]: typo? earlier was correct

File: python/fledge/common/iprpc.py [hunk 2]
```diff
@@ -235,23 +241,35 @@ def serve(self):
         """
 
         while True:
+            # eprint("RECEIVE: {}".format(str(_obj)[:200]))
+            try:
+                _obj = self.rpc_read()
+
```
Comment (praveen-garg) [inline python/fledge/common/iprpc.py:247]: better if newline avoided here :]

**Comment by deepanshu-yadav** [comment]
System tests are also passing. Ready to get merged.

**Review by deepanshu-yadav — APPROVED** [review]
Made some small changes in format and made DEBUG_RPC optional.

**Review by praveen-garg — APPROVED** [review]

File: python/fledge/common/iprpc.py [hunk 3]
```diff
@@ -18,14 +18,20 @@
...
 import time
 import logging
+
 import mmap
 
-from fledge.common.utils import eprint
+sys.path.append(os.path.dirname(__file__))
+
+DEBUG_RPC=False
```
[the remark this answers is not in the document]
Comment (doug-dianomic) [reply python/fledge/common/iprpc.py]: wow, how did I lose that?? the problem with having code in two different places :/

File: python/fledge/common/iprpc.py (hunk 2 above)
Comment (doug-dianomic) [reply python/fledge/common/iprpc.py]: I need to start running the linter you guys use. I'm trying to resist my preference to have more whitespace to highlight things, but my instincts fail me sometimes :)

**Pull request merged**